### PR TITLE
Add metrics collection

### PR DESCRIPTION
**Pull request by kevinwuTT**
### Problem description
Generate metrics that resembles the following example when running test models:
```
| Model   | Run Success | Torch Ops (Before)  | Torch Ops (Remains) | To/From_Device Ops | Original Run Time (s) | Compiled Run Time (s) | Accuracy          |
|---------|--------------|--------------------|---------------------|--------------------|-----------------------|-----------------------|-------------------|
| ResNet  | Yes          | 1234               | 567                 | 12                 | 0.45                  | 0.30                  | 85.5%             |
| VGG     | Yes          | 2345               | 890                 | 10                 | 0.50                  | 0.35                  | 92.1%             |
| BERT    | No           | 3456               | N/A                 | N/A                | N/A                   | N/A                   | N/A               |
```

### How to run
```
pytest tests/models
python3 tools/collect_metrics.py
```
`metrics.csv` will be generated in the base directory of the repo

Sample run on my Grayskull VM:
```
Model,Run Success,Torch Ops (Before),Torch Ops (Remain),To/From_Device Ops,Original Run Time (s),Compiled Run Time(s),Accuracy
ResNet18,Yes,70,43,132,1.7748399539850652,9.067756048985757,0.9999226429397775
BERT,Yes,1393,684,3862,61.74298016901594,38.48160677799024,0.9864120722326671
```

### What's changed
* Added util function that measures the runtime and status of a model. 
* Collect the count of torch ops before and after compilation.
* Collect the count of to/from_device ops after compilation.
* Create a helper script that parses the `metrics` directory for relevant files and generates a .csv that resembles the table above. The helper script also computes the accuracy between the original and compiled run by reading the relevant output files.
* Modify BERT and ResNet18 test models to request metrics collection

### CODE REVIEW

File: tests/models/bert/test_bert.py [hunk 1]
```diff
@@ -59,13 +59,13 @@ def decode_output(outputs):
         answer_before = decode_output(outputs_before)
 
         # Compile model with ttnn backend
-        option = torch_ttnn.TorchTtnnOption(device=self.device)
+        option = torch_ttnn.TorchTtnnOption(device=self.device, metrics_path="BERT")
```
Comment (ayerofieiev-tt) [inline tests/models/bert/test_bert.py:62]: we can move "BERT" constant

File: tests/models/bert/test_bert.py [hunk 2]
```diff
@@ -59,13 +59,13 @@ def decode_output(outputs):
...
         m = torch.compile(m, backend=torch_ttnn.backend, options=option)
 
         # Run inference with the compiled model
         with torch.no_grad():
-            outputs_after = m(**inputs)
-        option._out_fx_graphs[0].print_tabular()
+            outputs_after = RunTimeMetrics("BERT", "compiled", lambda: m(**inputs))
+        # option._out_fx_graphs[0].print_tabular()
```
Comment (ayerofieiev-tt) [inline tests/models/bert/test_bert.py:68]: remove?

Comment (kevinwuTT) [reply tests/models/bert/test_bert.py:68]: Should we actually keep this? This prints out the table view of the converted graph.

**Review by ayerofieiev-tt — APPROVED** [review]

File: tests/models/bert/test_bert.py (hunk 2 above)
Comment (ayerofieiev-tt) [reply tests/models/bert/test_bert.py:68]: We can keep it for now if it is useful
